web/hooks/use-follows-supabase.ts

```diff
@@ -0,0 +1,59 @@
+import { RealtimeChannel } from '@supabase/realtime-js'
+import { useEffect, useState } from 'react'
+import { db } from 'web/lib/supabase/db'
+import { getContractFollows } from 'web/lib/supabase/follows'
+import { useRealtimeChanges } from 'web/lib/supabase/realtime/use-realtime'
+import { useEffectCheckEquality } from './use-effect-check-equality'
+import { Delete, Insert } from 'common/supabase/realtime'
+import { uniq } from 'lodash'
+
+export function useRealtimeContractFollows(contractId: string | undefined) {
+  const [follows, setFollows] = useState<string[] | undefined | null>(undefined)
+
+  const deletedFollows: Delete<'contract_follows'>[] = useRealtimeChanges(
+    'DELETE',
+    'contract_follows',
+    {
+      k: 'contract_id',
+      v: contractId ?? '',
+    }
+  )
+  const insertedFollows: Insert<'contract_follows'>[] = useRealtimeChanges(
+    'INSERT',
+    'contract_follows',
+    {
+      k: 'contract_id',
+      v: contractId ?? '',
+    }
+  )
+
+  useEffectCheckEquality(() => {
+    if (!contractId) {
+      return
+    }
+    getContractFollows(contractId)
+      .then((result) => setFollows(result))
+      .catch((e) => console.log(e))
+  }, [contractId])
+
+  let combinedFollows = [
+    ...(follows ?? []),
+    ...insertedFollows.map((followData) => followData.new.follow_id),
+  ]
+
+  // Get array of follow_ids from deletedFollows
+  let deletedFollowIds = deletedFollows.map(
+    (deletedFollow) => deletedFollow.old.follow_id
+  )
+
+  // For each element in deletedFollows, find its index in combinedFollows.
+  // If it exists (the index is not -1), remove it from combinedFollows.
+  deletedFollowIds.forEach((deletedFollowId) => {
+    let index = combinedFollows.indexOf(deletedFollowId ?? '')
+    if (index !== -1) {
+      combinedFollows.splice(index, 1)
+    }
+  })
+
+  return uniq(combinedFollows)
```

Comment:
Unless I am missing something, I think this function could be replaced by
```typescript
const { rows } = useSubscription('contract_follows', { k: 'contract_id', v: contractId })
return rows?.map(r => r.follow_id)
```
As I wrote above, the fact that it accepts `undefined` is a code smell and indicates something is likely wrong with the React component design.